### ISSUE TITLE
test_tenant_relocation sometimes fails in the CI

### ISSUE THREAD

**Issue by LizardWizzard** [zenithdb/zenith#1193]
For example:

https://app.circleci.com/pipelines/github/zenithdb/zenith/4261/workflows/90684d8c-75de-44c6-a1e1-776b1c5bd2b3/jobs/39039/tests

Log of new pageserver contains following suspicious error:
```
2022-01-27T08:37:13.397122Z  INFO Starting a http endpoint at 127.0.0.1:15214
2022-01-27T08:37:13.404920Z  INFO POST /v1/timeline/74ee8b079a0e437eb0afea7d26a07209/27887d1a0f0940cee862ca936050f233/attach 202 Accepted
2022-01-27T08:37:13.438456Z  WARN Failed to read local metadata file, assuming it's safe to override its with the download. Read: Failed to read local metadata file bytes: No such file or directory (os error 2)
2022-01-27T08:37:16.895137Z ERROR Failed to download archive ArchiveId(0/1696268) (archives downloaded: 0; archives left: 1) for tenant 74ee8b079a0e437eb0afea7d26a07209 timeline 27887d1a0f0940cee862ca936050f233, requeueing the download: Failed to uncompress the archive

Caused by:
    0: Failed to uncompress archive entry FileEntry { size: 118943, subpath: RelativePath("rel_1663_12973_2617_0_0_0000000001696268") }
    1: Failed to write extracted archive contents into file /tmp/test_output/test_tenant_relocation[without_load]/repo/new_pageserver/tenants/74ee8b079a0e437eb0afea7d26a07209/timelines/27887d1a0f0940cee862ca936050f233/rel_1663_12973_2617_0_0_0000000001696268
    2: Failed to read exactly 118943 bytes from the input, bytes written: 0
2022-01-27T08:37:16.896810Z  INFO Updating states for 1 timelines    
2022-01-27T08:37:17.903115Z  WARN Failed to read local metadata file, assuming it's safe to override its with the download. Read: Failed to read local metadata file bytes: No such file or directory (os error 2)
2022-01-27T08:37:18.420211Z  INFO GET /v1/timeline/74ee8b079a0e437eb0afea7d26a07209/27887d1a0f0940cee862ca936050f233 200 OK
2022-01-27T08:37:19.428174Z  INFO GET /v1/timeline/74ee8b079a0e437eb0afea7d26a07209/27887d1a0f0940cee862ca936050f233 200 OK
```

Initially I thought that it may be caused by interleaving download with new checkpoint or gc, but I don't see any logs proving that theory. If I understand it correctly error says that archive was deleted before we were able to uncompress data from it.

@SomeoneToIgnore maybe you have some ideas on that

On my machine I cannot reproduce it. Continuing digging deeper

**Comment by SomeoneToIgnore** [comment]
The culprit is 

>     2: Failed to read exactly 118943 bytes from the input, bytes written: 0

meaning that during archive extraction, some file returned EOF instead of the complete set of file's bytes.
Since the whole archive is a streaming one, that would mean that the entire archive stream got EOF'ed somehow...
We require the archive stream directly from the S3, no intermediate files located, so that EOF could potentially mean that the server had gotten shut?

Here's another evidence that S3 mock server management might not be done properly, one of the `main` builds failed to connect to a mock server during the bucket: https://app.circleci.com/pipelines/github/zenithdb/zenith/4270/workflows/e549559a-7657-4586-936c-38fa0864b85d/jobs/39158
Yet the following `main` reruns did not fail on this test.

**Comment by LizardWizzard** [comment]
It seems test tenant relocation and test remote storage are both failing sometimes. While digging into test remote storage failure I've downloaded archives from CI and one of them has zero file size:

![image](https://user-images.githubusercontent.com/19991702/151421454-95d59c0d-caf1-4af1-b460-cc83ac068399.png)

I started inspecting upload logs for something unusual and noticed that shutdown was requested when archive was half way into remote storage:
```
2022-01-27T18:01:05.636687Z DEBUG Starting writing files into archive
2022-01-27T18:01:05.636761Z DEBUG Shutdown requested, stopping
``` 
So I think this is the reason. I'll increase waiting period to be sure everything landed in remote storage to check the hypothesis

**Comment by LizardWizzard** [comment]
The other discovered issue was already known, currently we allow to schedule download for timeline that is already scheduled for download, this leads to sync errors when we compare metadada and whole timeline eventually moved to evicted state which makes it unavailable for operations. #1163 should fix this behaviour

Regarding that zero sized archive which appeared after interrupted upload. This shouldnt happen with real s3 because it makes the file available after the upload is completed, so we do not get partial files after that. Probably we need something to workaround this in tests, also it maybe worth it ho have separate test case for real s3 th check that everything is as expected

**Comment by LizardWizzard** [comment]
Fixed by #1163
